media/js/base/stub-attribution.js

```diff
@@ -466,19 +466,31 @@ if (typeof window.Mozilla === 'undefined') {
     /**
      * Gets utm parameters and referrer information from the web page if they exist.
      * @param {String} ref - Optional referrer to facilitate testing.
+     * @param {Boolean} omitNonEssentialFields - Optional flag to omit fields that are nonEssential for RTAMO.
      * @return {Object} - Stub attribution data object.
      */
-    StubAttribution.getAttributionData = function (ref) {
+    StubAttribution.getAttributionData = function (
+        ref,
+        omitNonEssentialFields
+    ) {
         var params = new window._SearchParams();
         var utms = params.utmParams();
-        var experiment =
-            params.get('experiment') || StubAttribution.experimentName;
-        var variation =
-            params.get('variation') || StubAttribution.experimentVariation;
+        var experiment = omitNonEssentialFields
+            ? null
+            : params.get('experiment') || StubAttribution.experimentName;
+        var variation = omitNonEssentialFields
+            ? null
+            : params.get('variation') || StubAttribution.experimentVariation;
         var referrer = StubAttribution.getReferrer(ref);
-        var ua = StubAttribution.getUserAgent();
-        var clientIDUA = StubAttribution.getUAClientID();
-        var clientIDGA4 = StubAttribution.getGtagClientID();
+        var ua = omitNonEssentialFields
+            ? 'other'
```

Comment:
shouldn't we nullify the `ua` too (for consistency)?

Reply:
We could do, but we default to `other` in the case that we can't determine a user agent already. So I figured maybe that makes more sense?

Reply:
Oh so we never use `(notset)` for `ua`?